File: Models/ThanhToan.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace DoAnSE104.Models
{
    public class ThanhToan
    {
        [Key]
        public int MaThanhToan { get; set; }

        [Required]
        public int MaHoaDon { get; set; }
        [ForeignKey("MaHoaDon")]
        public HoaDon HoaDon { get; set; }

        [Required]
        public int MaNguoiThue { get; set; }
        [ForeignKey("MaNguoiThue")]
        public NguoiThue NguoiThue { get; set; }

        [Required]
        public DateTime NgayThanhToan { get; set; }

        [Required]
        [Column(TypeName = "decimal(18,2)")]
        public decimal TongTien { get; set; }

        [Required]
        [MaxLength(100)]
        public string HinhThucThanhToan { get; set; }

        [MaxLength(255)]
        public string? GhiChu { get; set; }
    }
} 

File: wwwroot/js/modules/thanh-toan.js
// Module cấu hình: thanhtoan
window.AppModules = window.AppModules || {};
window.AppModules.thanhtoan = {
        title: 'Thanh Toán',
        endpoint: '/api/ThanhToan',
        pk: 'maThanhToan',
        headers: [
            { label: 'Hóa đơn', key: 'maHoaDon', render: v => `HĐ#${v}` },
            { label: 'Khách thuê', key: 'maNguoiThue', render: v => lookups.nguoithue.find(n => n.maNguoiThue === v)?.hoTen || `#${v}` },
            { label: 'Ngày thanh toán', key: 'ngayThanhToan', render: fmtDate },
            { label: 'Số tiền', key: 'tongTien', render: fmtCurrency },
            { label: 'Hình thức', key: 'hinhThucThanhToan' },
            { label: 'Ghi chú', key: 'ghiChu' }
        ],
        fields: [
            { id: 'MaHoaDon', label: 'Hóa đơn', type: 'lookup', lookup: 'hoadon', valField: 'maHoaDon', txtField: 'maHoaDon', required: true },
            { id: 'MaNguoiThue', label: 'Khách thuê', type: 'lookup', lookup: 'nguoithue', valField: 'maNguoiThue', txtField: 'hoTen', required: true },
            { id: 'TongTien', label: 'Số tiền thanh toán (đ)', type: 'number', required: true },
            { id: 'HinhThucThanhToan', label: 'Hình thức thanh toán', type: 'lookup', lookup: 'hinhthuc', valField: 'val', txtField: 'label', required: true },
            { id: 'GhiChu', label: 'Ghi chú', type: 'text' }
        ]
    };
